### DISCUSSION TITLE
Difference between InstanceSplitter and Normal training ?

### DISCUSSION THREAD

**Discussion by mbrhd** [awslabs/gluonts#1641] [Q&A] [answered]
From other discussions, I understood that during training the training dataset is sliced out automatically by "training-windows" composed of context-length + prediction-length. 

I've also gone through the Extended Forecasting Tutorial where the InstanceSplitter function is used to generate both training-windows and test-windows.

What are the main features of "normal training" (randomly splitting that training dataset into windows)? Is my first assumption right?

In which use cases the InstancSplitter function is better to use?

Thank you

**Comment by mbohlkeschneider** [comment]
Hi @bidHimArm,

Thank you for the question. The `InstanceSplitter` implements "normal training" (i.e. randomly sampling windows from the time series and presenting it to the model). However, some models use different ways of passing the dataset (like MQ-CNN). 

Keeping the transformation configuration (including `InstanceSplitter`, where applicable) should be a reasonable choice in many cases. 

This this answer your question?

**Reply by mbrhd** [reply]
Hi @mbohlkeschneider 

Thank you very much for your quick response 🙏🏼

One more question. For the evaluation part, it looks like the `make_evaluation_predictions` takes the latest part (prediction length) of the test dataset to evaluate.  
Thus, if I want to evaluate my model on a longer history of the test set I'll need to use the `InstanceSplitter` function or a rolling forecast backtest like in the example bellow? 

[/examples/run_rolling_forecast_backtest.py](https://github.com/awslabs/gluon-ts/blob/master/examples/run_rolling_forecast_backtest.py)

Thanks

**Comment by mbohlkeschneider** [comment, marked as answer]
> Thus, if I want to evaluate my model on a longer history of the test set I'll need to use the InstanceSplitter function or a rolling forecast backtest like in the example bellow?

Sorry - I did not understand this. `make_evaluation_predictions` will evaluate the last `prediction_length` data points of the test set. There should be nothing to be done specifically for evaluation in the `InstanceSplitter`. If you follow the official examples, you should be fine.

**Reply by mbrhd** [reply]
Hi Michael

My concern here is to evaluate different windows of my test dataset. I'm using two years in the past (daily granularity) to build my training dataset, and I use the year after the two years (of training) as a test dataset.  So, I want to evaluate different windows if the test dataset and not only the last `prediction_length` data points of the test set.

In the [Extended Forecasting Tutorial](https://ts.gluon.ai/tutorials/forecasting/extended_tutorial.html) it `InstanceSplitter` function is applied to both training data and the tests dataset. Therefore, evaluation is done on the different windows of the test dataset. 

I've found another to evaluate different windows of the test set using a rolling forecast backtest (like a cross-validation) in this example: [/examples/run_rolling_forecast_backtest.py](https://github.com/awslabs/gluon-ts/blob/master/examples/run_rolling_forecast_backtest.py)

My question here is related to the `InstanceSplitter` function applied to the test dataset. Is right that this approach allows me to evaluate different windows of my test data set (in my case the year after my training dataset)?

Thank you very much for your support and patience

**Reply by mbohlkeschneider** [reply]
> My question here is related to the InstanceSplitter function applied to the test dataset. Is right that this approach allows me to evaluate different windows of my test data set (in my case the year after my training dataset)?

No, this is not correct. The way it is set up in the moment it will always extract the latest time points during inference . Using the rolling forecast backtest and/or constructing the rolling test dataset manually is recommended in this case.

**Reply by mbrhd** [reply]
Thanks for your support @mbohlkeschneider
